### DISCUSSION TITLE
Parsing multiple buffers

### DISCUSSION THREAD

**Discussion by OrangeLightning219** [nvim-treesitter/nvim-treesitter#3744] [General]
**Is your feature request related to a problem? Please describe.**
Currently treesitter parses only the active buffer. This makes it not very usefull for autocompletion or fuzzy finding (for example with telescope). 

**Describe the solution you'd like**
I would like treesitter to parse all files in a directory (or all open buffers) so that it can be used as an alternative to lsp.

**Describe alternatives you've considered**
I tried using lsp servers (clangd and ccls) but they break when using them with a unity build (single compilation unit) project.

I assume this wouldn't be hard to implement because syntax highlighting already works fine with splits. Maybe there is some configuration option I'm missing?

**Comment by clason** [comment]
Out of scope for this project; you can use the basic `vim.treesitter` functionality in core to do this (but a lot of assembly required).
